calibrate.py

```diff
@@ -9,128 +11,190 @@
 high_H = 0
 high_S = 0
 high_V = 0
-window_capture_name = 'Video Capture'
-window_detection_name = 'Object Detection'
-low_H_name = 'Lower Bound Offset H'
-low_S_name = 'Lower Bound Offset S'
-low_V_name = 'Lower Bound Offset V'
-high_H_name = 'Upper Bound Offset H'
-high_S_name = 'Upper Bound Offset S'
-high_V_name = 'Upper Bound Offset V'
+window_capture_name = "Video Capture"
+window_detection_name = "Object Detection"
+low_H_name = "Lower Bound Offset H"
+low_S_name = "Lower Bound Offset S"
+low_V_name = "Lower Bound Offset V"
+high_H_name = "Upper Bound Offset H"
+high_S_name = "Upper Bound Offset S"
+high_V_name = "Upper Bound Offset V"
+
+DEFAULT_PATH = "threshold.json"
+
+
+def save_threshold(path: str = DEFAULT_PATH):
+    dir_name = os.path.dirname(path)
+    if dir_name:
+        os.makedirs(dir_name, exist_ok=True)
+
+    data = {
+        "low_H": low_H,
+        "low_S": low_S,
+        "low_V": low_V,
+        "high_H": high_H,
+        "high_S": high_S,
+        "high_V": high_V,
+    }
+
+    with open(path, "w", encoding="utf-8") as f:
+        json.dump(data, f, indent=2, ensure_ascii=False)
+
+    return path
+
+
+def load_threshold(path: str = DEFAULT_PATH):
+    global low_H, low_S, low_V, high_H, high_S, high_V
+
+    if not os.path.exists(path):
+        save_threshold(path)
+
+    with open(path, "r", encoding="utf-8") as f:
+        data = json.load(f)
+
+    low_H = data["low_H"]
+    low_S = data["low_S"]
+    low_V = data["low_V"]
+    high_H = data["high_H"]
+    high_S = data["high_S"]
+    high_V = data["high_V"]
 
 
 def on_low_H_thresh_trackbar(val):
-	global low_H
-	global high_H
-	low_H = val
-	cv.setTrackbarPos(low_H_name, window_detection_name, low_H)
+    global low_H
```

Comment:
tabs, rebase on main and reformat. added .editorconfig so this happens automatically